STYLE.md

```diff
@@ -63,6 +63,46 @@ Don't make promises, even weak ones, about what we plan to do in the future, wit
 
 If your document needs to describe a feature that isn't implemented yet, either rewrite to not depend on that feature, or just say that it's a "current limitation" without elaborating further. Include a workaround if there is one.
 
+## Key vocabulary
+
+There are a small number of key terms that we need to use consistently throughout the documentation.
+
+The most important distinction is the following:
+
+- **Database**: This is the active, running entity that lives on a host. It contains a bunch of tables, like a normal database. It also has extra features: clients can connect to it directly and remotely call its stored procedures.
+- **Module**: This is the source code that a developer uses to specify a database. It is a combination of a database schema and a collection of stored procedures. Once built and published, it becomes part of the running database.
+
+A database **has** a module; the module **is part of** the database.
+
+The module does NOT run on a host. The **database** runs on a host.
+
+A client does NOT "connect to the module". A client **connects to the database**.
+
+This distinction is subtle but important. People know what databases are, and we should reinforce that SpacetimeDB is a database. "Module" is a quirky bit of vocabulary we use to refer to collections of stored procedures. A RUNNING APPLICATION IS NOT CALLED A MODULE.
+
+Other key vocabulary:
+- (SpacetimeDB) **Host**: the application that hosts **databases**. It is multi-tenant and can host many **databases** at once.
+- **Client**: any application that connects to a **database**.
+- **End user**: anybody using a **client**.
+- **Database developer**: the person who maintains a **database**.
```

Comment:
Are they database developers or module developers? In my mind, I see an analogy between "database" <-> "process" and "module" <-> "program." Following from that, developers write programs, not processes.

Reply:
"Database developer" is an industry standard term, although "database administrator" is more common. I think it applies.

Since we're deemphasizing the term "module" I think it's better to say "database" here.